grammar/haxe.bnf

```diff
@@ -490,8 +496,16 @@ left shiftExpression ::= shiftOperator additiveExpressionWrapper
 private additiveExpressionWrapper ::= multiplicativeExpressionWrapper additiveExpression*
 left additiveExpression ::= ('+' | '-') multiplicativeExpressionWrapper
 
-private multiplicativeExpressionWrapper ::= prefixExpression multiplicativeExpression*
-left multiplicativeExpression ::= ('*' | '/' | '%') (prefixExpression | suffixExpressionWrapper)
+private multiplicativeExpressionWrapper ::= assignExpressionWrapper multiplicativeExpression*
+left multiplicativeExpression ::= ('*' | '/' | '%') assignExpressionWrapper
+
+/* Technically, assignExpressions don't have higher priority than everything else.
+ * But for parsing, this makes sense; otherwise, assignment operators that comprise
```

Comment:
> If it could, I could simply "enter" a type declaration and tell the lexer not to make the shift tokens until a closing bracket is found.

Unfortunately, this wouldn't be quite sound either because type parameters can actually hold expressions. That is, something like `Array<[2 >> 1]>` is valid Haxe as far as the syntax is concerned. This would require keeping track of bracket scopes somehow, which I'm not sure you have control over.

Interesting article on the subject: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2005/n1757.html